extensions/src/main/java/org/pentaho/platform/plugin/action/mondrian/MondrianFlushCacheComponent.java

```diff
@@ -0,0 +1,76 @@
+/*!
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2002-2017 Pentaho Corporation..  All rights reserved.
+ */
+package org.pentaho.platform.plugin.action.mondrian;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.pentaho.platform.engine.services.solution.ComponentBase;
+import org.pentaho.platform.plugin.action.mdx.MDXLookupRule;
+import org.pentaho.platform.plugin.services.connections.mondrian.MDXConnection;
+
+import mondrian.olap.CacheControl;
+import mondrian.olap.Connection;
+import mondrian.olap.Cube;
+
+/**
+ * @author William E. Seyler
+ */
+public class MondrianFlushCacheComponent extends ComponentBase {
+
+  private static final long serialVersionUID = 5697771680582241114L;
+  private static final Log logger = LogFactory.getLog( MondrianFlushCacheComponent.class );
+
+  public boolean executeAction() {
+    MDXLookupRule mdxLookupRule = getLookupRule();
+    Connection conn = ( (MDXConnection) mdxLookupRule.shareConnection() ).getConnection();
+    CacheControl cacheControl = conn.getCacheControl( null );
+    Cube[] cubes = conn.getSchema().getCubes();
+    for ( Cube cube : cubes ) {
+      cacheControl.flush( cacheControl.createMeasuresRegion( cube ) );
+    }
+    cacheControl.flushSchema( conn.getSchema() );
+    return true;
+  }
+
+  @Override
+  protected boolean validateAction() {
+    return true;
```

Comment:
You need to verify the conditions you expect here. For example, what if the MDX Lookup Rule doesn't have a connection ? Then your code on line 39 will get an NPE, won't it ? So minimally, you need to make sure that there's a "shared_olap_connection" available for you to get in this method.